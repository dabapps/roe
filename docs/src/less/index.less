@import '../../../src/less/index.less';

@body-background: @grey-lighter;

h1,
h2 {
  color: @secondary;
}

.container {
  box-shadow: @shadow-hard;
}

.display-columns {
  .column {
    background-color: @grey-lightest;
    border: @border-base;
  }
}
