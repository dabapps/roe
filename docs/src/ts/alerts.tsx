import * as React from 'react';

import {
  Alert,
  CodeBlock,
  Column,
  DabIpsum,
  Row,
  Section
} from '../../../src/ts';

const Alerts = () => (
  <Section>
    <Row>
      <Column>
        <h2>
          Alerts
        </h2>

        <Alert>
          <DabIpsum count={2} />
        </Alert>
        <Alert className="info">
          <DabIpsum count={2} />
        </Alert>
        <Alert className="success">
          <DabIpsum count={2} />
        </Alert>
        <Alert className="warning">
          <DabIpsum count={2} />
        </Alert>
        <Alert className="error">
          <DabIpsum count={2} />
        </Alert>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Code
        </h3>
        <CodeBlock language="javascript" name="JSX">
          {`
            <Alert>
              <DabIpsum count={2} />
            </Alert>
            <Alert className="info">
              <DabIpsum count={2} />
            </Alert>
            <Alert className="success">
              <DabIpsum count={2} />
            </Alert>
            <Alert className="warning">
              <DabIpsum count={2} />
            </Alert>
            <Alert className="error">
              <DabIpsum count={2} />
            </Alert>
          `}
        </CodeBlock>
      </Column>
    </Row>
  </Section>
);

export default Alerts;
