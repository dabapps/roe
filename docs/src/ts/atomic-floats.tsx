import * as React from 'react';

import {
  CodeBlock,
  Column,
  Row,
  Section
} from '../../../src/ts';

const AtomicFloats = () => (
  <Section>
    <Row>
      <Column>
        <h2>
          Atomic Float Classes
        </h2>
      </Column>
    </Row>
    <Row>
      <Column className="display-columns">
        <h3>
          Demo
        </h3>
        <Row>
          <Column xs={6} className="xs-float-right sm-float-left md-float-right">
            float-right sm-float-left md-float-right
          </Column>
          <Column xs={6} className="xs-float-left sm-float-right md-float-left">
            float-left sm-float-right md-float-left
          </Column>
        </Row>
        <p>
          Note: float-left is an alias for xs-float-left, and float-right an alias for xs-float-right
        </p>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Code
        </h3>
        <CodeBlock language="javascript">
          {`
            <Row>
              <Column xs={6} className="xs-float-right sm-float-left md-float-right">
                float-right sm-float-left md-float-right
              </Column>
              <Column xs={6} className="xs-float-left sm-float-right md-float-left">
                float-left sm-float-right md-float-left
              </Column>
            </Row>
          `}
        </CodeBlock>
      </Column>
    </Row>
  </Section>
);

export default AtomicFloats;
