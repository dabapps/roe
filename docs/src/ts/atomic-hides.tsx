import * as React from 'react';

import {
  CodeBlock,
  Column,
  Row,
  Section
} from '../../../src/ts';

const AtomicHides = () => (
  <Section>
    <Row>
      <Column>
        <h2>
          Atomic Hide & Display Classes
        </h2>
      </Column>
    </Row>
    <Row>
      <Column className="display-columns">
        <h3>
          Demo
        </h3>
        <Row>
          <Column xs={6} className="xs-display-none sm-display-block md-display-none">
            xs-display-none sm-display-block md-display-none
          </Column>
          <Column xs={6} className="xs-display-block sm-display-none md-display-block">
            xs-display-block sm-display-none md-display-block
          </Column>
        </Row>
        <p>
          Available display types are: none, block, inline, inline-block
        </p>
        <p>
          Note: display-none is an alias for xs-display-none, and display-block an alias for xs-display-block, etc
        </p>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Code
        </h3>
        <CodeBlock language="javascript" name="JSX">
          {`
            <Row>
              <Column xs={6} className="xs-display-none sm-display-block md-display-none">
                xs-display-none sm-display-block md-display-none
              </Column>
              <Column xs={6} className="xs-display-block sm-display-none md-display-block">
                xs-display-block sm-display-none md-display-block
              </Column>
            </Row>
          `}
        </CodeBlock>
      </Column>
    </Row>
  </Section>
);

export default AtomicHides;
