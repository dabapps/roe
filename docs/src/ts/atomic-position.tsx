import * as React from 'react';

import {
  Anchor,
  CodeBlock,
  Column,
  Row,
  Section
} from '../../../src/ts';

const AtomicPosition = () => (
  <Section>
    <Row>
      <Column>
        <h2>
          <Anchor>
            Atomic Position Classes
          </Anchor>
        </h2>
      </Column>
    </Row>
    <Row>
      <Column className="display-columns">
        <h3>
          Demo
        </h3>
        <div className="atomic-position-demo">
          <p>This is a container div relatively positioned for purpose of this demo</p>
          <div className="position-relative">
            position-relative. relative positioning means the element is positioned relative to itself. Without using the top, right, bottom, or left properties, it behaves like a static element. Try adding properties to move this element!
          </div>
          <div className="position-absolute">
            position-absolute. absolute means positioned relative to its nearest positioned parent positioned element. It is now positioned relative to its nearest relative ancestor.
          </div>
        </div>
        <p>
          Note: all the padding and colour above was just part of the demo.
        </p>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Code
        </h3>
        <CodeBlock language="html" name="HTML">
          {`
            <div className="atomic-position-demo">
              <p>This is a container div relatively positioned for purpose of this demo</p>
              <div className="position-relative">
                position-relative. relative positioning means the element is positioned relative to itself. Without using the top, right, bottom, or left properties, it behaves like a static element. Try adding properties to move this element!
              </div>
              <div className="position-absolute">
                position-absolute. absolute means positioned relative to its nearest positioned parent positioned element. It is now positioned relative to its nearest relative ancestor.
              </div>
            </div>
          `}
        </CodeBlock>
      </Column>
    </Row>
  </Section>
);

export default AtomicPosition;
