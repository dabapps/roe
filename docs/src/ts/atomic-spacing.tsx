import * as React from 'react';

import {
  CodeBlock,
  Column,
  Row,
  Section
} from '../../../src/ts';

const AtomicSpacing = () => (
  <Section>
    <Row>
      <Column>
        <h2>
          Atomic Padding & Margin Classes
        </h2>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Demo
        </h3>
        <div className="padding-base">
          Padding all around
        </div>
        <div className="md-padding-left-large">
          Large padding left on medium screens and above
        </div>
        <div className="margin-left-base sm-margin-left-none">
          Margin left only only small screens
        </div>
        <div className="margin-vertical-small">
          Small vertical margin
        </div>
        <div className="margin-horizontal-large">
          Large horizontal margin
        </div>
        <p>
          Note: padding-base is an alias for xs-padding-base, and margin-right-small an alias for xs-margin-right-small,
          etc
        </p>
      </Column>
    </Row>
    <Row>
      <Column>
        <h3>
          Code
        </h3>
        <CodeBlock language="javascript" name="JSX">
          {`
            <div className="padding-base">
              Padding all around
            </div>
            <div className="md-padding-left-large">
              Large padding left on medium screens and above
            </div>
            <div className="margin-left-base sm-margin-left-none">
              Margin left only only small screens
            </div>
            <div className="margin-vertical-small">
              Small vertical margin
            </div>
            <div className="margin-horizontal-large">
              Large horizontal margin
            </div>
          `}
        </CodeBlock>
      </Column>
    </Row>
  </Section>
);

export default AtomicSpacing;
