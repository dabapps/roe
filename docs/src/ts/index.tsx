import * as React from 'react';
import * as ReactDOM from 'react-dom';

import './livereload';

import {
  Column,
  Container,
  Row
} from '../../../src/ts';

import { Buttons } from './buttons';
import { CodeBlocks } from './code-blocks';
import { Grid } from './grid';
import { Inputs } from './inputs';
import { Text } from './text';

const packageJson = require( '../../../package.json'); // tslint:disable-line:no-var-requires

class App extends React.Component<void, void> {
  public render () {
    return (
      <Container>
        <Row>
          <Column>
            <h1>
              Roe - DabApps' Project Development Kit (v{packageJson.version})
            </h1>
            <p>
              <strong>
                A Collection of React Components for Project Development
              </strong>
            </p>
          </Column>
        </Row>
        <Buttons />
        <CodeBlocks />
        <Grid />
        <Inputs />
        <Text />
      </Container>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('app'));
