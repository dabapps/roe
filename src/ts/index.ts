export { default as Alert } from './components/misc/alert';
export { default as AppRoot } from './components/app/root';
export { default as Anchor } from './components/misc/anchor';
export { default as Badge } from './components/misc/badge';
export { default as Banner } from './components/banners/banner';
export { default as Button } from './components/forms/button';
export { default as CodeBlock } from './components/misc/code-block';
export { default as Collapse } from './components/misc/collapse';
export { default as Column } from './components/grid/column';
export { default as Container } from './components/grid/container';
export { default as ContentBox } from './components/content/content-box';
export { default as ContentBoxHeader } from './components/content/content-box-header';
export { default as ContentBoxFooter } from './components/content/content-box-footer';
export { default as CookieBanner } from './components/banners/cookie-banner';
export { CookieBannerRenderProps } from './components/banners/cookie-banner';
export { default as DabIpsum } from './components/prototyping/dab-ipsum';
export { default as Footer } from './components/navigation/footer';
export { default as FormGroup } from './components/forms/form-group';
export { default as Highlight } from './components/misc/highlight';
export { default as InputGroup } from './components/forms/input-group';
export { default as InputGroupAddon } from './components/forms/input-group-addon';
export { default as InputWithPrefixSuffix } from './components/precomposed/input-with-prefix-suffix';
export { default as Modal } from './components/modals/modal';
export { default as ModalBody } from './components/modals/modal-body';
export { default as ModalCloseIcon } from './components/modals/modal-close-icon';
export { default as ModalFooter } from './components/modals/modal-footer';
export { default as ModalHeader } from './components/modals/modal-header';
export { default as ModalRenderer } from './components/modals/modal-renderer';
export { default as Nav } from './components/navigation/nav';
export { default as NavBar } from './components/navigation/nav-bar';
export { default as NavItem } from './components/navigation/nav-item';
export { default as Row } from './components/grid/row';
export { default as Section } from './components/content/section';
export { default as SideBar } from './components/navigation/side-bar';
export { default as SpacedGroup } from './components/misc/spaced-group';
export { default as SpeechBubble } from './components/misc/speech-bubble';
export { default as Tabs } from './components/tabs/tabs';
export { default as Tab } from './components/tabs/tab';
export { default as Table } from './components/tables/table';
export { default as TableBody } from './components/tables/table-body';
export { default as TableCell } from './components/tables/table-cell';
export { default as TableHead } from './components/tables/table-head';
export { default as TableHeader } from './components/tables/table-header';
export { default as TableRow } from './components/tables/table-row';
export { default as Well } from './components/misc/well';
