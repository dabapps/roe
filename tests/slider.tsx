import * as enzyme from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Slider } from '../src/ts/';

const props = {
  onChange: jest.fn(),
  min: 10,
  max: 333,
  initialValue: {
    from: 10,
    to: 333,
  },
  step: 5,
};

describe('Slider', () => {
  beforeAll(() => {
    jest.spyOn(window, 'addEventListener');
    jest.spyOn(window, 'removeEventListener');
    Element.prototype.getBoundingClientRect = jest.fn(() => {
      return { width: 100, height: 10, top: 0, left: 0, bottom: 0, right: 0 };
    });
  });

  beforeEach(() => {
    props.onChange.mockClear();
  });

  it('should match snapshot', () => {
    const instance = renderer.create(<Slider {...props} />);

    expect(instance).toMatchSnapshot();
  });

  it('should call mouse events on range stepped slider (first slide exceeded range position)', () => {
    const instance = enzyme.mount(<Slider {...props} />);

    instance
      .find('.control')
      .first()
      .simulate('mouseDown', { clientX: 20, clientY: 0 })
      .simulate('mouseup');

    instance
      .find('.control')
      .last()
      .simulate('mouseDown', { clientX: 80, clientY: 0 })
      .simulate('mouseup');

    const mouseMove = new MouseEvent('mousemove', { clientX: 20, clientY: 0 });
    const mouseUp = new MouseEvent('mouseup', { clientX: 80, clientY: 0 });

    mouseMove.initEvent('mousemove', true, true);
    mouseUp.initEvent('mouseup', true, true);

    document.dispatchEvent(mouseMove);
    document.dispatchEvent(mouseUp);

    instance.unmount();
    expect(props.onChange).toHaveBeenCalledTimes(2);
    expect(props.onChange).toHaveBeenCalledWith({ from: 75, to: 75 });
    expect(props.onChange).toHaveBeenCalledWith({ from: 75, to: 270 });
  });

  it('should call touch events on range stepped slider (first slide exceeded range position)', () => {
    const instance = enzyme.mount(<Slider {...props} />);

    instance
      .find('.control')
      .first()
      .simulate('touchmove', { targetTouches: [{ clientX: 10 }] })
      .simulate('touchend');

    instance
      .find('.control')
      .last()
      .simulate('touchmove', { targetTouches: [{ clientX: 10 }] })
      .simulate('touchend');

    const touchMove = new TouchEvent('touchmove', {
      targetTouches: [{ clientX: 400 } as Touch],
    });
    const touchEnd = new TouchEvent('touchend', { targetTouches: [] });

    touchMove.initEvent('touchmove', true, true);
    touchEnd.initEvent('touchend', true, true);

    document.dispatchEvent(touchMove);
    document.dispatchEvent(touchEnd);

    instance.unmount();
    expect(props.onChange).toHaveBeenCalledTimes(2);
    expect(props.onChange).toHaveBeenCalledWith({ from: 40, to: 333 });
    expect(props.onChange).toHaveBeenCalledWith({ from: 40, to: 333 });
  });

  it('should remove listeners on unmount', () => {
    const instance = enzyme.shallow(<Slider {...props} />);

    instance.unmount();
  });

  it('should avoid mouse right click', () => {
    const instance = enzyme.mount(<Slider {...props} />);

    instance
      .find('.control')
      .first()
      .simulate('mouseDown', { button: 2 })
      .simulate('mouseMove', { clientX: 100, clientY: 0 })
      .simulate('mouseUp');

    instance.unmount();
    expect(props.onChange).not.toHaveBeenCalled();
  });
});
