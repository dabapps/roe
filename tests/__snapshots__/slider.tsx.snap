// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should match snapshot 1`] = `
<div
  step={5}
>
  <div
    className="slider"
  >
    <div
      className="bar"
    >
      <span
        className="background"
      />
      <div
        className="control"
        onMouseDown={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      >
        <span
          className="control-popover"
        >
          <span>
            10
          </span>
        </span>
      </div>
      <span
        className="range"
        style={
          Object {
            "left": "0%",
            "right": "0%",
          }
        }
      />
      <div
        className="control"
        onMouseDown={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "left": "100%",
          }
        }
      >
        <span
          className="control-popover"
        >
          <span>
            333
          </span>
        </span>
      </div>
      <span
        className="background"
      />
    </div>
  </div>
  <p
    className="margin-vertical-small"
  >
    <span>
      10
    </span>
    <span
      className="float-right"
    >
      333
    </span>
  </p>
</div>
`;
